Shortening MachineShopSimulator.java

smit9025 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
I moved a section from the method moveToNextMachine() into its own method for readability, and because I had extracted and moved it into another class before but decided to move it back for now. I might unextract it into its parent method unless I find that I can move it. I moved a section from the method changeState() in MachineShopSimulator.java to Machine.java since it interacted more with Machine.java and named it processWork. I encapsulated three fields in MachineShopSimulator.java so that they can be gotten in other methods. I moved setUpJobs from MachineShopSimulator.java to SimulationSpecification.java. Originally MSS object was being passed to the moved methods, but I changed it to take more specific parameters instead. I also changed some uses of "this" to be more specific. 

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- Please link to the issue here: -->
Issue #12

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- Describe the smell this addresses and the particular location the smell occurs -->
<!--- Provide a summary of the description in the issue, but don’t necessarily copy all the details over to the pull request. -->
<!--- Make sure to include enough information that people don’t have to click over the issue if they aren’t inclined. -->
<!--- You need to make sure the issue has the necessary detail. -->
This change is required because it reduces the large class smell by reducing methods inside of it. It also reduces data class smell by giving some methods to other classes that they should be in instead of MachineShopSimulator.
Issue #12 is reducing large class smell in machineshopsimulator.java.

## Smell impact
<!--- What smells does this reduce/eliminate? -->
<!--- What smells does this introduce/increase? -->
<!--- Were there old smells that you hadn’t noticed that this helps bring to your attention? -->
This reduces the large class and data class smells. It doesn't introduce/increase any other smell as far as I can tell. I didn't discover any new smells.

## Testing & Screenshots:
<!--- Screenshot of the code coverage report so we can track that over time. -->
<!--- Mention any new tests here as well. -->
<img width="959" alt="CoverageReport" src="https://user-images.githubusercontent.com/46760134/96355925-549e9e00-10ad-11eb-9f73-0962f240b30b.PNG">

No new tests were made.
- [x] All new and existing tests passed.

smit9025 committed:
Move a section from the method moveToNextMachine() in MachineShopSimulator.java to Job.java since it interacted more with Job.java and named it scheduleTask.
Move a section from the method changeState() in MachineShopSimulator.java to Machine.java since it interacted more with Machine.java and named it processWork.
Encapsulate three fields in MachineShopSimulator.java so that they can be get in other methods.

smit9025 committed:
Move setUpJobs from MachineShopSimulator.java to SimulationSpecification.java

UtkarshKr007 reviewed (approved):
I think this refactoring works for shortening the MSS class. I do think, this should be **followed up right away** by refactoring that makes it so those classes don't have to take in the entire MSS class as a parameter. Additionally, this also creates circular dependency between MSS having an instance of the machine to call its method and then the machine class having an instance of the MSS class.

deglm006 commented:
It is not entirely clear to me how each of the issues are related to each other. I feel like each issue might be better in it's own PR.
Also, I feel like the names of the issues are not great. They don't really inform a reviewer exactly what that issue is supposed to be addressing.

smit9025 commented:
I did some editing of issues and removed a few so that hopefully it is more apparent what is going on with the issues. I'm sorry for the confusion, we have been meaning to redo the issues and make them a lot better than the mess that we currently have going on that I created early on with not much effort. Utkarsh also brings up a good point that I'd like to address in what would probably be the next pull request. I would like to be able to not pass the whole machineshopsimulator object for a function that only uses a couple of things from that class. I've done a little tinkering around and have found a couple of things that work so far, but will have to do further work on that.

BananaObserved commented on src/applications/MachineShopSimulator.java:
```diff
@@ -29,15 +29,7 @@ boolean moveToNextMachine(Job theJob, SimulationResults simulationResults) {
-                changeState(p);
-            }
-            return true;
+            return theJob.scheduleTask(this);
```
If you're passing `this` as an argument, it feels like that's a sign that the logic's gone upside-down somewhere.  It might be better to just pass in the fields that the function would need?  Although, that's a lot of them.  Perhaps this is the best way of handling things... there isn't an obvious elegant solution.

BananaObserved commented on src/applications/MachineShopSimulator.java:
```diff
@@ -194,4 +150,28 @@ public static void main(String[] args) {
     public int getTimeNow() {
         return timeNow;
     }
+
+    public EventList geteList() {
+        return eList;
+    }
+
+    public void seteList(EventList eList) {
+        this.eList = eList;
+    }
+
+    public Machine[] getMachine() {
+        return machine;
+    }
+
+    public void setMachine(Machine[] machine) {
+        this.machine = machine;
+    }
+
+    public int getLargeTime() {
+        return largeTime;
+    }
+
+    public void setLargeTime(int largeTime) {
+        this.largeTime = largeTime;
```
This looks like inappropriate intimacy to me.  Aren't all of these private, or at least implicitly private?

BananaObserved reviewed (commented):
There's definitely some good and useful stuff here, and it makes some sense to move logic out where it belongs.  That being said, it feels like some inappropriate intimacy is occuring, as the MachineShopSimulator is simply passing itself to other functions, and those other functions are accessing many of its internal fields.  It might make more sense to only pass in the specifically necessary fields--or, if that's most of them, leave it in MachineShopSimulator.  That being said, a lot of the changes (such as `processWork`) at least _seem_ to make a lot of sense. and could probably just be passed a few fields?  Having to get an internal list and set to it does seem ugly, though; ideally you could simply do that through the MachineShopSimulator.

smit9025 committed:
Move scheduleTask back into MachineShopSimulator since it was using more data from machineshopsimulator than it was using within Job.java. Inappropriate intimacy was created by moving it to Job.java.
Removed use of "this" within MachineShopSimulator.java since Ben brought up that it might not be the best idea to be passing more data than strictly specified.
Removed use of machineshopsimulator parameter from beginNextJob() and processWork() in Machine.java because it was raised by Utkarsh that it might not be the best idea to pass the whole data object of MSS instead of being specific.
Remove set<var>() functions in MSS since they weren't being used anymore.
Changed setUpJobs() so that it takes a Machine[] instead of MSS object.
